File: arcade/python/arcade-intro/05_Island of Knowledge/020_arrayMaximalAdjacentDifference.py
def arrayMaximalAdjacentDifference(inputArray):
    '''
    Given an array of integers, find the maximal absolute difference between any two of its adjacent elements.
    '''
    
    if len(inputArray) < 2:
        raise "The inputArray must contain at least 2 elements"
        
    maxDiff = abs(inputArray[0] - inputArray[1])
    
    for i in range(len(inputArray)-1):
        diff = abs(inputArray[i] - inputArray[i+1])
        if diff > maxDiff:
            maxDiff = diff
    return maxDiff

print(arrayMaximalAdjacentDifference([2, 4, 1, 0]))

File: arcade/go/arcade-intro/05_Island of Knowledge/020_arrayMaximalAdjacentDifference.go
func arrayMaximalAdjacentDifference(inputArray []int) int {
	maxDif := 0
	for i := 0; i < len(inputArray)-1; i++ {
		dif := inputArray[i] - inputArray[i+1]
		if dif < 0 {
			dif = -dif
		}

		if dif > maxDif {
			maxDif = dif
		}
	}

	return maxDif
}
